docs/admin/federation/index.md

```diff
@@ -0,0 +1,103 @@
+
+```shell
+$ FEDERATION=true KUBE_RELEASE_RUN_TESTS=n make quick-release
+```
+
+Next, you need to push the images to a registry from where your kubernetes cluster can read them.
+If your kubernetes cluster is running on GCE, then you can push the images to `gcr.io/<your-gce-project-name>`.
+The command to push the images will look like:
+
+```shell
+$ FEDERATION=true FEDERATION_PUSH_REPO_BASE=gcr.io/<your-gce-project-name> ./build/push-federation-images.sh
+```
+
+### Running the federation control plane
+
+Federation control plane consists of an etcd, federation-apiserver and federation-controller-manager.
+You can run these binaries as pods on your existing kubernetes cluster.
+The command to run these pods on an existing GCE cluster will look like:
+
+```shell
+$ KUBERNETES_PROVIDER=gce FEDERATION_PUSH_REPO_BASE=gcr.io/<your-gce-project-name> ./federation/cluster/federation-up.sh
+```
+
+This creates a namespace `federation` and creates 2 deployments: `federation-apiserver` and `federation-controller-manager`.
+You can verify that the pods are available by running the following command:
+
+```shell
+$ kubectl get deployments --namespace=federation
+NAME                            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+federation-apiserver            1         1         1            1           10m
+federation-controller-manager   1         1         1            1           10m
+```
+
+Running `federation-up.sh` also creates a new record in your kubeconfig for you
+to be able to talk to federation apiserver. You can view this by running
+`kubectl config view`.
+
+## Registering kubernetes clusters for federation
+Now that we have the federation control plane up and running, we can start registering kubernetes clusters.
+
+First of all, we need to create a secret containing kubeconfig for that kubernetes cluster, which federation control plane will use to talk to that kubernetes cluster.
+For now, we create this secret in the host kubernetes cluster (that hosts federation control plane). When we start supporting secrets in federation control plane, we will create this secret their.
+Suppose that your kubeconfig for kubernetes cluster is at /cluster1/kubeconfig, you can run the following command to create the secret:
+
+```shell
+$ kubectl create secret generic cluster1 --namespace=federation --from-file=/cluster1/kubeconfig
+```
+
+Note that the file name should be `kubeconfig` since file name determines the name of the key in the secret.
+
+Now that the secret is created, we are ready to register the cluster. The YAML file for cluster will look like:
+
+```yaml
+apiVersion: v1alpha1
+kind: Cluster
+metadata:
+  name: cluster1
+spec:
+  serverAddressByClientCIDRs:
+  - clientCIDR: <client-cidr>
+    serverAddress: <apiserver-address>
+  secretRef:
+    name: <secret-name>
+```
+
+You need to insert the appropriate values for `<client-cidr>`, `<apiserver-address>` and `<secret-name>`.
```

Comment:
What does the `client-cidr` refer to here?

If I have a cluster from another region I want to add (with its own load balancer), where would I find the CIDR?